### PR TITLE
Mdht release 2.0 merge

### PR DESCRIPTION
**Pull request by shereenElSayed**
I created cmake files for apps/raft , apps/dht, apps/monitor, and any subdirectories.
I updated some c files by removing the #include woofc-host, any caching calls,..
There are some problems:
--important packages need to be installed for raft and dht (shell scripts in these folders are available to install them)
--monitor cmake folder has a problem (the file causing it is commented for now)
--I am having a problem building the apps/dht/app/multi-regression with finding boost, but this might be a problem with my machine and not a general problem

### CODE REVIEW

**Comment by shereenElSayed** [comment]
@FatihBAKIR I am having some issues with the libraries (boost, mlpack). I specified the path for boost in the code to make it work. I am sure there is a more neat way to do this. for mlpack although it is installed, I am still getting error for undefined funcion mlpack::regression::linear_regression...

File: makefile [hunk 1]
```diff
@@ -0,0 +1,117 @@
+CC=gcc
```
Comment (FatihBAKIR) [inline makefile:1]: We don't do makefiles anymore :)

File: include/woofc.h [hunk 2]
```diff
@@ -27,6 +28,11 @@ int WooFValidURI(const char* str);
 
 int WooFLocalIP(char* ip_str, int len);
 
+unsigned long WooFGetElSize(WOOF* wf, const char* wf_name);
+
+char* WoofGetFileName(WOOF* wf);
```
Comment (FatihBAKIR) [inline include/woofc.h:33]: This should return a `const char*`

File: include/woofc.h [hunk 3]
```diff
@@ -27,6 +28,11 @@ int WooFValidURI(const char* str);
 
 int WooFLocalIP(char* ip_str, int len);
 
+unsigned long WooFGetElSize(WOOF* wf, const char* wf_name);
```
Comment (FatihBAKIR) [inline include/woofc.h:31]: Why do we need both the woof pointer and its name?

File: src/woofc.cpp [hunk 4]
```diff
@@ -82,7 +82,7 @@ int WooFCreate(const char* name, unsigned long element_size, unsigned long histo
             memset(local_name, 0, sizeof(local_name));
             strncpy(local_name, WooF_dir, sizeof(local_name));
             if (local_name[strlen(local_name) - 1] != '/') {
-                strncat(local_name, "/", 1);
+                strncat(local_name, "/", 2);
```
Comment (FatihBAKIR) [inline src/woofc.cpp:85]: Could you explain what this is fixing?

File: apps/monitor/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,20 @@
+
+add_library(monitor monitor.c)
+target_link_libraries(monitor PUBLIC woof )
+target_include_directories(monitor PUBLIC "${CMAKE_SOURCE_DIR}/apps/monitor/include")
```
[the remark this answers is not in the document]
Comment (shereenElSayed) [reply apps/monitor/CMakeLists.txt:4]: Yup changing that

File: apps/dht/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,107 @@
+
+set(OPENSSL_USE_STATIC_LIBS TRUE)
+set(OPENSSL_MSVC_STATIC_RT TRUE)
+set(OPENSSL_INCLUDE_DIR "/usr/local/ssl/include")
+set(OPENSSL_LIBRARIES "/usr/local/ssl/lib")
+set(OPENSSL_ROOT_DIR "/usr/local/ssl")
+find_package(OpenSSL REQUIRED)
+pkg_search_module(OPENSSL REQUIRED openssl)
```
Comment (FatihBAKIR) [inline apps/dht/CMakeLists.txt:8]: Why do we have some hard-coded paths?

File: apps/dht/apps/multi_regression/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,26 @@
+
+FIND_PACKAGE(Boost REQUIRED)
```
Comment (FatihBAKIR) [inline apps/dht/apps/multi_regression/CMakeLists.txt:2]: Use lower-case for consistency (ie. `find_package(Boost REQUIRED)`)

File: src/net/zmq/client.cpp [hunk 8]
```diff
@@ -185,6 +185,7 @@ int32_t backend::remote_get_elem_size(std::string_view woof_name_v) {
     }
 
     auto& [str] = *res;
+    DEBUG_LOG("%s ", str.c_str());
```
Comment (FatihBAKIR) [inline src/net/zmq/client.cpp:188]: This will cause a lot of logs

**Review by FatihBAKIR — CHANGES_REQUESTED** [review]
There are only a few changes needed, looking good

File: makefile (hunk 1 above)
Comment (shereenElSayed) [reply makefile:1]: I will delete this file. I was keeping the makefiles just in case of any failures and we need to take a look on them.

Comment (shereenElSayed) [reply makefile:1]: Shall I delete all makefiles in /apps?

File: include/woofc.h (hunk 3 above)
Comment (shereenElSayed) [reply include/woofc.h:31]: Because in case of remote, we use `WooFMsgGetElSize(wf_name)` otherwise if shared, we return `wf->shared->element_size`

**Pull request merged**